rageval/metrics/_answer_ter.py

```diff
@@ -0,0 +1,158 @@
+    'AnswerCorrectness'
+    >>> s, ds = metric.compute(dataset)
+    >>> type(ds)
+    <class 'datasets.arrow_dataset.Dataset'>
+"""
+
+_CITATION = """\
+@inproceedings{snover-etal-2006-study,
+    title = "A Study of Translation Edit Rate with Targeted Human Annotation",
+    author = "Snover, Matthew  and
+      Dorr, Bonnie  and
+      Schwartz, Rich  and
+      Micciulla, Linnea  and
+      Makhoul, John",
+    booktitle = "Proceedings of the 7th Conference of the Association for Machine Translation in the Americas: Technical Papers",
+    month = aug # " 8-12",
+    year = "2006",
+    address = "Cambridge, Massachusetts, USA",
+    publisher = "Association for Machine Translation in the Americas",
+    url = "https://aclanthology.org/2006.amta-papers.25",
+    pages = "223--231",
+}
+@inproceedings{post-2018-call,
+    title = "A Call for Clarity in Reporting {BLEU} Scores",
+    author = "Post, Matt",
+    booktitle = "Proceedings of the Third Conference on Machine Translation: Research Papers",
+    month = oct,
+    year = "2018",
+    address = "Belgium, Brussels",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/W18-6319",
+    pages = "186--191",
+}
+"""
+
+
+@dataclass
+@add_attribute('mtype', 'AnswerCorrectness')
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class AnswerTERCorrectness(Metric):
+    """Estimates the TER between answers and ground truth answers."""
+
+    name = "answer_ter"
+
+    ALIAS = ['answer_ter']
+
+    def __init__(self, model: str = "en_core_web_sm"):
+        """
+        Explicitly initialize AnswerTERCorrectness.
+
+        Ensure all parent classes are initialized.
+        """
+        super().__init__()
+        self._required_columns = ['answers', 'gt_answers']
+
+    def __repr__(self) -> str:
+        """:return: Formatted string representation of the metric."""
+        return f"{self.ALIAS[0]}"
+
+    def _info(self):
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            citation=_CITATION,
+            features=datasets.Features(
+                {
+                    "answers": datasets.Value("string"),
+                    "gt_answers": datasets.Value("string")
+                }
+            )
+        )
+
+    def _validate_data(self, dataset: datasets.Dataset) -> bool:
+        """Validate the of the input dataset."""
+        super()._validate_data(dataset)
+        if not all(isinstance(answer, str) for answer in dataset["answers"]):
+            raise ValueError("The type of answers should be a string.")
+        if not all(isinstance(a, List) or not all(isinstance(item, str) for item in a) for a in dataset["gt_answers"]):
+            raise ValueError("The type of gt_answers should be a list of strings.")
+
+    def compute(
+        self,
+        dataset: Dataset,
+        normalized: bool = False,
```

Comment:
Do not change the api of the function "compute", and you can remove most of the parameters into "__init__" as needed.

Besides, you should re-implement the function "_compute_batch", not the function "_compute".

Reply:
I remove most of the parameters into "init". 
In ter or chrf metric, the final score is not the average of scores of each sample, so there is a need to change the function "_compute"